pyproject.toml

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+
+# License: BSD 3-Clause
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "openml"
+dynamic = ["version"]  # Will take it from the __version__ file, update there
+dependencies = [
+  "liac-arff>=2.4.0",
+  "xmltodict",
+  "requests",
+  "scikit-learn>=0.18",
+  "python-dateutil",  # Installed through pandas anyway.
+  "pandas>=1.0.0",
+  "scipy>=0.13.3",
+  "numpy>=1.6.2",
+  "minio",
+  "pyarrow",
+]
+requires-python = ">=3.6"
+authors = [
+  { name = "Matthias Feurer", email="feurerm@informatik.uni-freiburg.de" },
+  { name = "Jan van Rijn" },
+  { name = "Arlind Kadra" },
+  { name = "Pieter Gijsbers" },
+  { name = "Neeratyoy Mallik" },
+  { name = "Sahithya Ravi" },
+  { name = "Andreas Müller" },
+  { name = "Joaquin Vanschoren " },
+  { name = "Frank Hutter" },
+]
+readme = "README.md"
+description = "Python API for OpenML"
+classifiers = [
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Programming Language :: Python",
+  "Topic :: Software Development",
+  "Topic :: Scientific/Engineering",
+  "Operating System :: POSIX",
+  "Operating System :: Unix",
+  "Operating System :: MacOS",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.6",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+]
+license = { file = "LICENSE" }
+
+[project.scripts]
+openml = "openml.cli:main"
+
+[project.optional-dependencies]
+test=[
```

Comment:
think about removing stuff after switching to ruff

Reply:
Seems that flake8 was never actually defined to be installed. Might have been installed through another package implicitly.

Either way, I think `flake8` is the only thing that is replaced by ruff, while adding a lot more.